src/webots/nodes/WbCamera.cpp

```diff
@@ -107,16 +107,12 @@ void WbCamera::init() {
   mExposure = findSFDouble("exposure");
 
   // backward compatibility
-  const WbSFString *type = findSFString("type");
-  if (type->value().startsWith('r', Qt::CaseInsensitive))
-    parsingWarn("Range finder type is not available for camera since Webots 8.4, please use a RangeFinder node instead.");
-
-  WbSFDouble *colorNoise = findSFDouble("colorNoise");
-  if (colorNoise->value() != 0.0) {  // Introduced in Webots 8.4.0
-    parsingWarn("Deprecated 'colorNoise' field, please use the 'noise' field instead.");
-    if (mNoise->value() == 0.0)
-      mNoise->setValue(colorNoise->value());
-    colorNoise->setValue(0.0);
+  WbSFBool *sphericalField = findSFBool("spherical");
+  if (sphericalField->value()) {  // Introduced in Webots R2023
+    parsingWarn("Deprecated 'spherical' field, please use the 'projection' field instead.");
+    if (isPlanarProjection())
+      mProjection->setValue("cylindrical");
+    sphericalField->setValue(false);
```

Comment:
Why do you set it to false here? If the user saves, it will be FALSE while they didn't changed anything...

Reply:
This is the backward compatibility method.
When loading the node we want to convert the settings of the deprecated `spherical` field to the new `projection` field.
So we exactly want to reset the `spherical` field because should no longer be used.
Even if the user doesn't change nothing, the saved world should be now compatible with R2023c and the `projection` field should be set and the `spherical` field should be reset to default.

Reply:
OK, that makes sense. Thank you for the explanation.